eclipse-theia/theia
# GH-7347: Added scroll-lock to the Output view

#### What it does


 - Adds a scroll-lock to the Output view,
 - Removes the `@theia/output` dependency from the `@theia/languages` extension,
 - Replaces the Output view with a `monaco` editor,
 - Adds context menu to the Output view,
 - Removes the `<no output yet>` when there is no output on the selected channel,
 - Fixes a runtime error when creating an in-memory (no URI) `monaco.editor.ITextModel`,
 - Can add decoration via `append` (previously it was supported with `appendLine` only),
 - Aligns the `OutputChannel` API with VS Code (via commands), and
 - Adds quick pick for _showing_, _hiding_, and _closing_ and output channel.

#### How to test

 - You can lock the scrolling in the view,
 - You can use the context menu from the Output view,
 - You can clear the Output view,
 - You can show, hide and close output channels from the _Command Palette_,
 - Clicking on the tab or in the widget sets the focus properly on the widget. You can see the bluish focus color (with the dark theme) when it's active. Also, there are no warnings due to widget did not activate in the logs (#7008),
 - You can alter the max output line count from the preferences, the output scales, (set it to e.g. 20)
 - Once you are done with the testing, make sure you dispose one of the output channels (use the `Output: Close Output Channel...` command), refresh the browser, make a memory snapshot, verify that the previously disposed channel is not in the snapshot.

 Feel free using the interim commands for populating the output channels with some dummy inputs:
   - `API Sample: Post Date.now() to the 'X' channel.`
   - `API Sample: Stop Date.now() on 'X' channel.`

Closes #7347.
Closes #7008.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
